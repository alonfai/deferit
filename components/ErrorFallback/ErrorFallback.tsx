import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

export type Props = {
  /**
   * Error Boundary exception object
   */
  error: Error;
  /**
   * Reload the app invoke method
   */
  resetError: () => void;
};

const ErrorFallback: React.FC<Props> = ({ error, resetError }) => {
  return (
    <View style={styles.container}>
      <MaterialIcons name='error' size={24} color='red' />
      <Text style={styles.heading}>Something went wrong:</Text>
      <Text style={styles.message}>{error.message}</Text>
      <Button title='Try Again' accessibilityLabel='Try Again' onPress={resetError}>
        Try again
      </Button>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  heading: {
    marginVertical: 10,
  },
  message: {
    color: 'red',
  },
});

export default ErrorFallback;
