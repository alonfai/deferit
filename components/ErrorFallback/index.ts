export { default } from './ErrorFallback';
export type { Props } from './ErrorFallback';
