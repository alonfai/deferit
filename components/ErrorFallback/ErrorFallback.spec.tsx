/// <reference types="@testing-library/jest-native" />
import React from 'react';
import { render } from '@testing-library/react-native';

import ErrorFallback from './ErrorFallback';

describe('<ErrorFallback />', () => {
  test('Render successfully', async () => {
    const error = new Error('error');

    const mockResetError = jest.fn();
    const { getByText, getByA11yLabel } = render(
      <ErrorFallback error={error} resetError={mockResetError} />
    );
    const genericError = getByText('Something went wrong:');
    const errorElement = getByText('error');
    const button = getByA11yLabel('Try Again');
    expect(genericError).toBeTruthy();
    expect(errorElement).toBeTruthy();
    expect(button).toBeTruthy();
  });
});
