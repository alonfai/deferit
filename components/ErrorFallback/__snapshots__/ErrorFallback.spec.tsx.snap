// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorFallback /> Render successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    Something went wrong:
  </Text>
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
    error
  </Text>
  <View
    accessibilityLabel="Try Again"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Try Again
      </Text>
    </View>
  </View>
</View>
`;
